### DISCUSSION TITLE
Trying to setup online backend

### DISCUSSION THREAD

**Discussion by Nemzytch** [strapi/strapi#7966] [Legacy Topics]
Hello, i've just followed this tutorial :

https://www.netlify.com/blog/2020/08/11/deploy-a-strapi-and-react-blog-on-netlify/

wich explains how to setup a strapi app online with heroku and cloudinary. This is cool we thought we could have an online backend. But the thing is, we cant actually manage collections online, we have to pull the heroku folder locally and push it everytime we want to make a new collection.

Is there any other way to have a strapi app running online "simply " ? We tryed with the node.js module of " o2switch " wich didnt really gave any good result  ( we didnt manage to get it running to be honest. )

If any of of you ever managed to setup a " fully " online backed, it would be nice to share some tips, thank in advance !

**Comment by derrickmehaffy** [comment]
When you are dealing with "serverless" style deployments (Heroku, Google App Engine, Azure App Engine) in which you push an update into some kind of source control and this triggers an automated deployment, you will run into this a lot unless you are dealing with a VPS (Virtual private server aka virtual machine). 

Strapi was not intended to be used like a monolithic CMS (see wordpress/drupal/ect) wherein you update, create, or delete content types in "production" (we forcefully disable this). You should be developing locally, pushing into source control, either auto/manually deploy into a staging environment for testing, then pushing into production.

You can see a break down of this flow here: https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#why-can-t-i-create-or-update-content-types-in-production-staging

If you want to go against our advice and "test in production" so to speak, then I would recommend you look into a VM/VPS and maybe add in your own open-source PaaS application layer (which is what something like Heroku is, PaaS).

A good example of this would be: https://github.com/dokku/dokku or https://github.com/flynn/flynn

**Reply by Nemzytch** [reply]
Thanks a lot, it's a bit beyond my actuall skill but then i'll spend some night putting at at my level of reach. :')  At least I have some stuff to dig around.
